server/deskroom/knowledge_base/endpoints.py

```diff
@@ -19,8 +29,59 @@ async def get_knowledge_base(
 ) -> list[KnowledgeBase]:
     response = (
         await supabase.table("knowledge_base")
-        .select("*")
+        .select("*, organizations(company_info_policy)")
         .eq("org_key", item.organization_key)
         .execute()
     )
+
     return response.data
+
+
+@router.post("/upload/{org_key}")
+async def make_knowledge_base(
+    org_key: str,
+    file: UploadFile = File(...),
+    supabase: AsyncClient = Depends(create_supabase_async_client),
+) -> list[KnowledgeBase]:
+    questions = []
+    answers = []
+
+    raw_df = pd.read_excel(
+        file.file.read(), engine="openpyxl", sheet_name="Message data"
+    )
+
+    supabase_response = await (
+        supabase.table("organizations").select("*").eq("key", org_key).execute()
+    )
+    org_info = supabase_response.data[0]
+
+    raw_df = raw_df.dropna()
+    processed_df = await process_raw_file(raw_df)
+
+    discovery_str = await generate_discovery_string(processed_df)
+    company_policy = await create_policy(discovery_str)
+    chat_ids = list(processed_df["chatId"].unique())
+    for chat_id in chat_ids:
+        try:
+            qa_string = await generate_qa_string(processed_df, chat_id)
+            discovered = await create_qa(company_policy, qa_string)
+            discovered_ = literal_eval(discovered)
+            for qa in list(discovered_.values()):
+                questions.append(qa["Question"])
+                answers.append(qa["Answer"])
+        except BaseException:
+            logger.info("Retrieved Messages")
+            pass
+    update_df = pd.DataFrame({"Question": questions, "Answer": answers})
+    updated_data = []
+    for _ in range(len(update_df)):
```

Comment:
```suggestion
    for row_num in range(len(update_df)):
```
_는 파이썬에서 사용하지 않는 변수를 구조화해야할 때 사용합니다. 변수 하나하나에 읽는 사람을 고려해주세요.